std/assembly/math.ts

```diff
@@ -2298,6 +2525,101 @@ export namespace NativeMathf {
     }
     return sx ? -x : x;
   }
+
+  export function sincos(x: f32): void { // see: musl/tree/src/math/sincosf.c
+    const s1pio2 = reinterpret<f64>(0x3FF921FB54442D18); // 1 * M_PI_2
+    const s2pio2 = reinterpret<f64>(0x400921FB54442D18); // 2 * M_PI_2
+    const s3pio2 = reinterpret<f64>(0x4012D97C7F3321D2); // 3 * M_PI_2
+    const s4pio2 = reinterpret<f64>(0x401921FB54442D18); // 4 * M_PI_2
+
+    var ix = reinterpret<u32>(x);
+    var sign = ix >> 31;
+    ix &= 0x7fffffff;
+
+    if (ix <= 0x3f490fda) {  /* |x| ~<= π/4 */
+      if (ix < 0x39800000) { /* |x| < 2**-12 */
+        sincos_s32 = x;
+        sincos_c32 = 1;
+        return;
+      }
+      sincos_s32 = sin_kernf(x);
+      sincos_c32 = cos_kernf(x);
+      return;
+    }
+
+    if (ASC_SHRINK_LEVEL < 1) {
```

Comment:
I wonder if this implies that results will diiffer slightly depending on shrink level?

Reply:
no, result is the same. This just fast paths which allow skip argument reduction stage

Reply:
I see, looks like a good optimization to me then.